src/components/shared/ListView/ListView.tsx

```diff
@@ -0,0 +1,212 @@
+    } else {
+      response = await service(params.queryParams)
+    }
+    setData(response)
+  }, [service, params])
+
+  useEffect(() => {
+    if (isReady) {
+      fetchData()
+    }
+  }, [fetchData, isReady])
+
+  const viewModeToggle = useMemo(() => {
+    return (
+      <ButtonGroup isAttached variant="outline">
+        <IconButton
+          aria-label="Grid View"
+          isActive={viewMode === "grid"}
+          icon={<HiOutlineViewGrid />}
+          onClick={() => setViewMode("grid")}
+        />
+        <IconButton
+          aria-label="List View"
+          isActive={viewMode === "table"}
+          icon={<HiOutlineViewList />}
+          onClick={() => setViewMode("table")}
+        />
+      </ButtonGroup>
+    )
+  }, [viewMode])
+
+  const handleUpdateQuery = useCallback(
+    (queryKey: string) => (value: string | boolean | number) => {
+      push({pathname: pathname, query: {...query, [queryKey]: value}})
+    },
+    [push, pathname, query]
+  )
+
+  const metaInformationDisplay = useMemo(() => {
+    if (!data) return
+    return (
+      <Text
+        alignSelf="center"
+        flexShrink={0}
+        fontWeight="bold"
+      >{`${data.Meta.ItemRange[0]} - ${data.Meta.ItemRange[1]} of ${data.Meta.TotalCount}`}</Text>
+    )
+  }, [data])
+
+  const renderList = useMemo(() => {
+    if (data) {
+      if (data.Items.length) {
+        const mergedGridOptions = {...DEFAULT_GRID_OPTIONS, ...gridOptions}
+        return (
+          <Box mb={5}>
+            {viewMode === "grid" ? (
+              //GRID VIEW
+              <Grid as="section" {...mergedGridOptions} w="full" width="100%">
+                {data.Items.map((o, i) => (
+                  <GridItem
+                    colSpan={1}
+                    rowSpan={1}
+                    bg="gridCellBg"
+                    w="full"
+                    width="100%"
+                    rounded="lg"
+                    key={i}
+                    borderStyle="none"
+                  >
+                    {renderCard(o, i, selected.includes(o.ID), handleSelectChange)}
+                  </GridItem>
+                ))}
+              </Grid>
+            ) : (
+              //TABLE VIEW
+              data.Items.map((o, i) => (
+                <pre key={i}>
+                  {JSON.stringify({ID: o.ID, Name: o.Name, isSelected: selected.includes(o.ID)}, null, 2)}
+                </pre>
+              ))
+            )}
+            {/* PAGINATION */}
+          </Box>
+        )
+      }
+      //NO RESULTS DISPLAY
+    }
+    //NONE CREATED DISPLAY
+  }, [data, viewMode, gridOptions, renderCard, selected, handleSelectChange])
+
+  const childrenProps = useMemo(() => {
+    return {
+      viewModeToggle,
+      metaInformationDisplay,
+      updateQuery: handleUpdateQuery,
+      routeParams: params.routeParams,
+      queryParams: params.queryParams,
+      children: renderList
```

Comment:
I thought that using that phrasing would make them think they use it like <listComponent/> instead of {listComponent} - which is why I initially had renderList. Just realized I never undid my change making that map to `children`. I agree that's confusing.